robot/src/main/java/us/ilite/robot/modules/Elevator.java

```diff
@@ -0,0 +1,280 @@
+
+package us.ilite.robot.modules;
+
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMax.IdleMode;
+import com.team254.lib.util.Units;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.team254.lib.drivers.talon.TalonSRXFactory;
+
+import us.ilite.common.config.SystemSettings;
+import us.ilite.lib.drivers.SparkMaxFactory;
+
+import edu.wpi.first.wpilibj.Talon;
+import edu.wpi.first.wpilibj.Timer;
+import us.ilite.common.lib.control.PIDController;
+import us.ilite.common.lib.control.PIDGains;
+import com.team254.lib.util.Util;
+import us.ilite.common.Data;
+
+
+public class Elevator extends Module {
+
+    private Data mData;
+    private boolean mAtBottom = true;
+    private boolean mAtTop = false;
+    private int mCurrentEncoderTicks;
+    private double mDesiredPower = 0;
+    private boolean mAtDesiredPosition;
+    private boolean mDesiredDirectionUp, mDesiredDirectionDown;
+    private boolean mDesiredPositionAboveInitial = false;
+    private double mMaxPower = 1.0d;
+    private double mMinPower = -1.0d;
+    private double mHoldVoltage = 0.5; //NOT final
+    private boolean mSettingPosition = false;
+    private int kCansparkId = 15; //TODO change ID
+    private int kTalonId = 15; //TODO change ID
+    private PIDController mPidController;
+    private double mCurrentTime;
+
+    //TODO need to figure out these encoder thresholds
+    private int mDownEncoderThreshold = SystemSettings.kLowerElevatorEncoderThreshold;
+    private int mUpEncoderThreshold = SystemSettings.kLowerElevatorEncoderThreshold;
+
+    EElevatorPosition mCurrentPosition = EElevatorPosition.BOTTOM;
+
+    EElevatorState mCurrentState = EElevatorState.NORMAL;
+    EElevatorPosition mDesiredPosition;
+
+    CANSparkMax mMasterElevator;
+    TalonSRX mEncoder = TalonSRXFactory.createDefaultTalon(kTalonId);
+
+    public Elevator(Data pData) {
+
+        this.mData = pData;
+        PIDGains pidGains = new PIDGains( SystemSettings.kElevatorP, SystemSettings.kElevatorI, SystemSettings.kElevatorD );
+        this.mPidController = new PIDController( pidGains, SystemSettings.kELevatorControlLoopPeriod );
+
+        //Create default NEO and set the ramp rate
+        mMasterElevator = SparkMaxFactory.createDefaultSparkMax(kCansparkId, MotorType.kBrushless);
+        mMasterElevator.setIdleMode(IdleMode.kBrake);
+        mMasterElevator.setRampRate(SystemSettings.kELevatorControlLoopPeriod);
+
+
+        //We start at the bottom
+        mAtBottom = true;
+        mAtTop = false;
+        mDesiredDirectionUp = true;
+
+        //Make sure the elevator is stopped upon initialization
+        mDesiredPosition = EElevatorPosition.BOTTOM;
+        mCurrentState = EElevatorState.STOP;
+
+        mCurrentEncoderTicks = 0;
+    }
+
+    public void shutdown(double pNow) {
+
+    }
+
+    public void modeInit(double pNow) {
```

Comment:
modeInit() should re-set the P, I, D, and F, mMinPower and mMaxPower values based upon system settings.  That way we can send new setting's via the display, dis-able then re-enable the robot with the new settings.